File: Backend/src/main/java/com/luckyseven/backend/core/LoggingInterceptor.java
package com.luckyseven.backend.core;

import com.luckyseven.backend.sharedkernel.exception.ErrorResponse;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.stereotype.Component;
import org.springframework.web.servlet.HandlerInterceptor;
@Component
public class LoggingInterceptor implements HandlerInterceptor {
  @Override
  public boolean preHandle(HttpServletRequest request,
      HttpServletResponse response,
      Object handler) {
    System.out.println("=========  Incoming Request Data =========");
    System.out.println("Request URL    : " + request.getRequestURL());
    System.out.println("Request Method : " + request.getMethod());
    System.out.println("Query Params   : " + request.getQueryString());
    return true;
  }

  @Override
  public void afterCompletion(HttpServletRequest request,
      HttpServletResponse response,
      Object handler,
      Exception ex) {
    System.out.println("=========  Outgoing Response Data =========");
    System.out.println("Response Status : " + response.getStatus());
    System.out.println("Request URL      : " + request.getRequestURL());
    System.out.println("Request Method   : " + request.getMethod());
    System.out.println("Query Params     : " + request.getQueryString());

    //  request 에 담긴 ErrorResponse 꺼내기
    Object obj = request.getAttribute("errorResponse");
    if (obj instanceof ErrorResponse err) {
      System.out.println("Error Code    : " + err.getCode());
      System.out.println("Error Message : " + err.getMessage());
      System.out.println("Error Detail  : " + err.getDetail());
    }
  }
}

File: Backend/src/main/kotlin/com/luckyseven/backend/core/LoggingInterceptor.kt
package com.luckyseven.backend.core

import com.luckyseven.backend.sharedkernel.exception.ErrorResponse
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.stereotype.Component
import org.springframework.web.servlet.HandlerInterceptor

@Component
class LoggingInterceptor : HandlerInterceptor {
    
    override fun preHandle(
        request: HttpServletRequest,
        response: HttpServletResponse,
        handler: Any
    ): Boolean {
        println("=========  Incoming Request Data =========")
        println("Request URL    : ${request.requestURL}")
        println("Request Method : ${request.method}")
        println("Query Params   : ${request.queryString}")
        return true
    }
    
    override fun afterCompletion(
        request: HttpServletRequest,
        response: HttpServletResponse,
        handler: Any,
        ex: Exception?
    ) {
        println("=========  Outgoing Response Data =========")
        println("Response Status : ${response.status}")
        println("Request URL      : ${request.requestURL}")
        println("Request Method   : ${request.method}")
        println("Query Params     : ${request.queryString}")
        
        // request 에 담긴 ErrorResponse 꺼내기
        val obj = request.getAttribute("errorResponse")
        if (obj is ErrorResponse) {
            println("ErrorResponse found in request attributes")
            println("ErrorResponse object: $obj")
        }
        
        // Exception이 있는 경우 로깅
        if (ex != null) {
            println("Exception occurred: ${ex.message}")
        }
    }
} 